### PR TITLE
Update Python and NPM dependencies

### PR DESCRIPTION
**Pull request by zsoltparal-mollie**
This MR updates all NPM and Python dependencies to the newest version that we can support. The two notable exceptions are:

1. ESLint, as the configuration has ^7.0.0 as the peer dependency, so I kept it in that range instead of going for v8
2. Sphinx 7.0.0, which included a lot more breaking changes than 6.0.0, so I went with the easier upgrade first

### CODE REVIEW

File: source/reference/v2/subscriptions-api/update-subscription.rst [hunk 1]
```diff
@@ -215,9 +215,6 @@ Response
 .. code-block:: json
    :linenos:
 
-    HTTP/1.1 200 OK
-    Content-Type: application/hal+json
```
Comment (zsoltparal-mollie) [inline source/reference/v2/subscriptions-api/update-subscription.rst:219]: This file failed to build with `code-block: http` for some reason. All other code blocks worked, but this one would keep failing.

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-sphinx==1.8.6
+sphinx==6.2.1
 jinja2==3.0.0
+sphinx-autobuild==2021.3.14
 urllib3==1.*
```
Comment (mvdpanne) [inline requirements.txt:4]: NBC: I think it is likely that we now no longer need to pin to version 1.x of this package. This was related to being 'stuck' on such an old version of sphinx. The same counts for the `jinja2` package.

Comment (zsoltparal-mollie) [reply requirements.txt:4]: I tried removing these, but the build fails without them.

Comment (mvdpanne) [reply requirements.txt:4]: Too bad, thanks for trying, though! 😄

**Comment by zsoltparal-mollie** [comment]
It seems like our Sphinx theme is not compatible with the latest version and it seems to be unmaintained. We should think about how to proceed.

**Pull request closed without merging**